app/search/models.py

```diff
@@ -42,6 +45,135 @@ def get_latest_test_results():
 
     return _get_latest_test_results(normalized_date)
 
+class DatasetCache(object):
+    def __init__(self, current_app):
+        self.current_app = current_app
+        dataset_cache_dir = current_app.config['DATASET_CACHE_PATH']
+        if not os.path.exists(dataset_cache_dir):
+            os.makedirs(dataset_cache_dir)
+
+    @property
+    def maxSize(self):
+        config_value = self._parseSize(self.current_app.config['DATASET_CACHE_MAX_SIZE'])
+        available_space = shutil.disk_usage(self.current_app.config['DATASET_CACHE_PATH']).used
+        return min(available_space - self.usedSpace , config_value)
+
+    @property
+    def usedSpace(self):
+        return sum(f.stat().st_size for f in os.scandir(self.current_app.config['DATASET_CACHE_PATH']) if (f.is_file))
+
+    @property
+    def freeSpace(self):
+        return self.maxSize - self.usedSpace
+        
+    @property
+    def cachedDatasets(self):
+        return dict(
+            (f.name, f)
+            for f in os.scandir(self.current_app.config['DATASET_CACHE_PATH'])
+        )
+
+    def getZippedContent(self, dataset):
```

Comment:
should take version. But I don't know how to get a previous version of a dataset. 
By commit# or tag. Are we using those?